frontend_tests/zjsunit/namespace.js

```diff
@@ -396,21 +399,55 @@ exports.with_overrides = function (test_function) {
         // code can also use this, as needed.)
         new_f._patched_with_override = true;
 
-        if ("__esModule" in obj && "__Rewire__" in obj) {
-            obj.__Rewire__(func_name, new_f);
-            restore_callbacks.push(() => {
-                obj.__Rewire__(func_name, old_f);
-            });
-        } else {
-            obj[func_name] = new_f;
-            restore_callbacks.push(() => {
-                obj[func_name] = old_f;
-            });
+        obj[func_name] = new_f;
+        restore_callbacks.push(() => {
+            obj[func_name] = old_f;
+        });
+    };
+
+    const override_rewire = function (obj, func_name, f) {
+        // This is deprecated because it relies on the slow
+        // babel-plugin-rewire-ts plugin.  Consider alternatives such
+        // as exporting a helper function for tests from the module
+        // containing the function you need to mock.
+
+        if (typeof f !== "function") {
+            throw new TypeError(
+                "You can only override with a function. Use with_field for non-functions.",
+            );
+        }
+
+        if (typeof obj !== "object" && typeof obj !== "function") {
+            throw new TypeError(`We cannot override a function for ${typeof obj} objects`);
         }
+
+        if (obj[func_name] !== undefined && typeof obj[func_name] !== "function") {
+            throw new TypeError(`
+                You are overriding a non-function with a function.
+                This is almost certainly an error.
```

Comment:
For Tim's benefit note that splitting out a separate function here allows us to streamline things a bit.  For example, in `override_wire` there is no need to set ` new_f._patched_with_override = true;`, and, in fact, it would lead to an anti-pattern.  All the superficial duplication between these two functions are pretty boilerplate in nature.